crawler/code/crawling.py

```diff
@@ -199,44 +199,48 @@ def fetch(self, url, max_redirect):
                                                  num_new_urls=0))
             return
 
-        if is_redirect(response):
-            location = response.headers['location']
-            next_url = urllib.parse.urljoin(url, location)
-            self.record_statistic(FetchStatistic(url=url,
-                                                 next_url=next_url,
-                                                 status=response.status,
-                                                 exception=None,
-                                                 size=0,
-                                                 content_type=None,
-                                                 encoding=None,
-                                                 num_urls=0,
-                                                 num_new_urls=0))
-
-            if next_url in self.seen_urls:
-                return
```

Comment:
Ah! I knew there was a case where the response wasn't released, but I didn't see it before now.